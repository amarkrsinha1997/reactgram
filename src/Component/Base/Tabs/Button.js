import React from 'react';
import PropTypes from 'prop-types';
import { colors } from '../../../styles/colors';

const style =  `
.tab {
	background: ${colors.grey.shade1};
	padding: 10px 20px;
	color: ${colors.black.shade1};
	text-align: center;
}
`
const Button = (props) => {
	return (
		<div className="tab" role="button" onClick={props.onClick}>
			{props.children}
			<style>{style}</style>
		</div>
	)
}

Button.propTypes = {
	onClick: () => {}
}
export { Button }

