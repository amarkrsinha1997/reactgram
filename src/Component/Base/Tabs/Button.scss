.tab {
  border: none;
  color: #333;
  padding: 10px 20px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  cursor: pointer;
  background: #dedede;
  text-align: center;
  width: 100%;
  box-shadow: 0px 1px 3px 1px rgba(33,33,33,0.3);
  &:hover {
    background: rgba(156,156,156, 0.3);
  }
  &:active {
    background: rgba(156,156,156, 0.8);
    box-shadow: 3px 5px 2px 1px rgba(156,156,156, 0.3);
  }
  
}