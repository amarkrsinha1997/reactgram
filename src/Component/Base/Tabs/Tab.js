import React from 'react';
import { colors } from '../../../styles/colors';

const style =  `
.tab {
	background: ${colors.grey.shade1};
	padding: 10px 20px;
	color: ${colors.black.shade1};
	text-align: center;
}
`
const Tab = (props) => {
	return (
		<div className="tab">
			{props.children}
			<style>{style}</style>
		</div>
	)
}

export { Tab }

