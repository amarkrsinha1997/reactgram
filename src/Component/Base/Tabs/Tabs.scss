.tabs {
	display: flex;
	justify-content: space-around;
	width: 100%;
}
.seperator {
	border-right: 1px solid #999;
	flex: 1;
	&:last-child {
		border-right: none;
	}
}