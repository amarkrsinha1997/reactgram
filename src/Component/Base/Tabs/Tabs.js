import React, { useMemo } from 'react'
import { Button } from './Button';
import './Tabs.scss'

function Tabs(props) {
	const children = useMemo(() => {
		const result = []
		React.Children.forEach(props.children, child => {
			const type = child && child.type && (child.type.displayName || child.type.name);
			if ((Button.displayName || Button.name).includes(type)) {
        result.push(<div className="seperator">{child}</div>)
			}
		})
		return result
	}, [props.children])

	return (
		<div className="tabs">
			{children}
		</div>
	)
}
Tabs.Button = Button;
export { Tabs }

