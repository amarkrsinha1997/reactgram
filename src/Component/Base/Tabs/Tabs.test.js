import React from 'react';
import { render, screen } from '@testing-library/react';
import { Tabs } from './Tabs';
import { Tab } from './Tab';
describe('Tabs', () => {

	test('renders Tabs properly', () => {
		const wrapper = <Tabs>
			<Tab>
				Like 1
			</Tab>
			<Tab>
				Comment 1
			</Tab>
		</Tabs>
		const { container } = render(wrapper);
		expect(container).toMatchSnapshot();
	});

	test('renders not render unwanted child', () => {
		const wrapper = <Tabs>
			<Tab>
				Like 1
            </Tab>
			<div data-testid="unwanted">
				Comment 1
            </div>
		</Tabs>
		const { container } = render(wrapper);
		const result = screen.queryByText(/comment 1/);
		expect(result).toBeFalsy();
		expect(container).toMatchSnapshot();
	});
});
