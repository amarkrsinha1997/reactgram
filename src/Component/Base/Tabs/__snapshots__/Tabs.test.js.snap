// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders Tabs properly 1`] = `
<div>
  <div
    class="tabs"
  >
    <div
      class="seperator"
    >
      <button
        class="tab"
      >
        Like 1
      </button>
    </div>
    <div
      class="seperator"
    >
      <button
        class="tab"
      >
        Comment 1
      </button>
    </div>
  </div>
</div>
`;

exports[`Tabs renders not render unwanted child 1`] = `
<div>
  <div
    class="tabs"
  >
    <div
      class="seperator"
    >
      <button
        class="tab"
      >
        Like 1
      </button>
    </div>
  </div>
</div>
`;
